### PR TITLE
Operator

### PR DESCRIPTION
**Pull request by 0xtekgrinder**

### CODE REVIEW

**Comment by 0xtekgrinder** [comment]
Still needs to update vault addresses on deploy

File: operator/src/utils/getParaswapData.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  destToken: string,
+  destDecimals: string,
+  amount: BigNumber,
+  userAddress: string,
+  chainId: number,
+  slippage: number
+): Promise<string> {
+  try {
+    const priceRoute: any = await axios.get(
+      `https://apiv5.paraswap.io/prices?srcToken=${srcToken}&destToken=${destToken}&amount=${amount.toString()}&side=SELL&network=${chainId}&srcDecimals=${srcDecimals}&destDecimals=${destDecimals}&userAddress=${userAddress}`
```
Comment (0xMemoryGrinder) [inline operator/src/utils/getParaswapData.ts:16]: This should be in a configuration attribute

File: operator/src/utils/getParaswapData.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    );
+    if (!priceRoute["priceRoute"]) {
+      throw new Error("No price route found");
+    }
+
+    const priceData: any = await axios.post(
+      `https://apiv5.paraswap.io/transactions/${chainId}?ignoreChecks=true&ignoreGasEstimate=true`,
```
Comment (0xMemoryGrinder) [inline operator/src/utils/getParaswapData.ts:23]: This should be in a configuration attribute

File: operator/src/utils/getParaswapData.ts (hunk 1 above)
Comment (0xMemoryGrinder) [inline operator/src/utils/getParaswapData.ts:16]: You should use URL class to build the url, it is cleaner to list all query parameters

**Review by 0xMemoryGrinder — CHANGES_REQUESTED** [review]
Check comments

File: operator/src/utils/getParaswapData.ts (hunk 1 above)
Comment (0xtekgrinder) [reply operator/src/utils/getParaswapData.ts]: overpowered

File: operator/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      } catch (err: any) {
+        console.log(`Harvest failed: ${err.message}`);
+        retryHarvestRunning = false;
+      }
+    },
+    { scheduled: false }
+  );
+
+  let retryCompoundRunning = false;
+  const retryCompoundTask = cron.schedule(
```
[the remark this answers is not in the document]
Comment (0xtekgrinder) [reply operator/src/index.ts:58]: I don't think it's needed

File: operator/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    try {
+      await compound(vaultAddress, maxGasPrice, slippage);
+    } catch (err: any) {
+      console.log(`Compound failed: ${err.message}`);
+      retryCompoundTask.start();
+    }
+  });
+
+  let retryHarvestRunning = false;
+  const retryHarvestTask = cron.schedule(
```
[the remark this answers is not in the document]
Comment (0xtekgrinder) [reply operator/src/index.ts:37]: I don't think it's needed

File: operator/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    throw new Error("STAKER_ADDRESS is not defined");
+  }
+
+  const slippage = parseInt(process.env.SLIPPAGE);
+  const maxGasPrice = parseFloat(process.env.MAX_GAS_PRICE);
+  const vaultAddress = process.env.VAULT_ADDRESS;
+  const stakerAddress = process.env.STAKER_ADDRESS;
+
+  let running = false;
+  cron.schedule("0 4 * * 0", async () => {
```
[the remark this answers is not in the document]
Comment (0xtekgrinder) [reply operator/src/index.ts:13]: I don't think it's needed

File: operator/src/utils/checkGasPrice.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import config from "../config/config";
+
+const checkGasPrice = async (maxGasPrice: number): Promise<number> => {
+  const etherscanApiKey = config.etherscanApiKey();
+
+  let gasPrice: number;
+  try {
+    const result: any = await axios.get(
+      `${etherscanApiKey}/api?module=gastracker&action=gasoracle&apikey=${etherscanApiKey}`
+    );
```
Comment (0xMemoryGrinder) [inline operator/src/utils/checkGasPrice.ts:11]: Use URL class to add query parameters

File: operator/src/utils/getParaswapData.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  destDecimals: string,
+  amount: BigNumber,
+  userAddress: string,
+  chainId: number,
+  slippage: number
+): Promise<string> {
+  const paraswapApiUrl = config.paraswapApiUrl();
+  try {
+    const priceRoute: any = await axios.get(
+      `${paraswapApiUrl}/prices?srcToken=${srcToken}&destToken=${destToken}&amount=${amount.toString()}&side=SELL&network=${chainId}&srcDecimals=${srcDecimals}&destDecimals=${destDecimals}&userAddress=${userAddress}`
```
Comment (0xMemoryGrinder) [inline operator/src/utils/getParaswapData.ts:18]: Use URL class to add query parameters

File: operator/src/utils/getParaswapData.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    );
+    if (!priceRoute["priceRoute"]) {
+      throw new Error("No price route found");
+    }
+
+    const priceData: any = await axios.post(
+      `${paraswapApiUrl}/transactions/${chainId}?ignoreChecks=true&ignoreGasEstimate=true`,
```
Comment (0xMemoryGrinder) [inline operator/src/utils/getParaswapData.ts:25]: Use URL class to add query parameters

**Review by 0xMemoryGrinder — CHANGES_REQUESTED** [review]
Check comments

**Review by 0xMemoryGrinder — APPROVED** [review]
LGTM

**Pull request merged**